asset/src/partition_by_date/processor.ts

```diff
@@ -0,0 +1,39 @@
+import { BatchProcessor, DataEntity } from '@terascope/job-components';
+import { PartitionByDateConfig } from './interfaces';
+
+export default class PartitionByDate extends BatchProcessor<PartitionByDateConfig> {
+    addPath(record: DataEntity) {
+        const partitions: string[] = [];
+        const offsets = {
+            daily: 10,
+            monthly: 7,
+            yearly: 4
+        };
+        // This value is enforced by the schema
+        const end = offsets[this.opConfig.resolution];
+        const dates = new Date(record[this.opConfig.field]).toISOString().slice(0, end).split('-');
```

Comment:
We should use `getValidDate` from `@terascope/utils` to handle edge cases with the date on the record.